From a9d027ee07f56b67054ec8ef02481a1e60429f6d Mon Sep 17 00:00:00 2001
From: Karel Zak <kzak@redhat.com>
Date: Wed, 10 Jul 2024 10:29:20 +0200
Subject: login: lastlog create

This is port of the old RHEL patch to RHEL-10.

Signed-off-by: Karel Zak <kzak@redhat.com>
---
 login-utils/login.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/login-utils/login.c b/login-utils/login.c
index c8544f6a1..3de0398e9 100644
--- a/login-utils/login.c
+++ b/login-utils/login.c
@@ -681,7 +681,7 @@ static void log_lastlog(struct login_context *cxt)
 	sa.sa_handler = SIG_IGN;
 	sigaction(SIGXFSZ, &sa, &oldsa_xfsz);
 
-	fd = open(_PATH_LASTLOG, O_RDWR, 0);
+	fd = open(_PATH_LASTLOG, O_RDWR | O_CREAT, 0);
 	if (fd < 0)
 		goto done;
 	offset = cxt->pwd->pw_uid * sizeof(ll);
-- 
2.45.2

